src/coordinator.coffee

```diff
@@ -97,9 +97,10 @@ pingUrl = (address, method, callback) ->
 
 class Coordinator extends EventEmitter
   constructor: (@broker, @options = {}) ->
-    @participants = {}
+    @participants = {} # participantId -> Definition (from discovery)
     @connections = {} # connId -> { queue: opt String, handler: opt function }
     @iips = {} # iipId -> { metadata, data }
+    @nodes = {} # role -> { metadata: {} }
```

Comment:
I would just go and add anything discovered to nodes. So then the graph is:

* Things that graph designer wanted to be there (whether they're currently running or not)
* _In addition_ other things that are running

It can be a bit confusing to have the differentiation between:

* Processes you can start and stop by adding/removing them from the graph (things coordinator manages)
* Processes that are essentially out of your control (things coordinator ignores but send discovery)

But that can't be helped. Maybe we can pass some metadata to signify the difference, since for example in the latter case trying to remove such node will be no-op.